packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2930,6 +2932,17 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return expr;
       }, state);
 
+      // report error if single type parameter used without trailing comma.
+      if (
+        arrow.node?.type === "ArrowFunctionExpression" &&
+        arrow.node?.typeParameters?.params.length === 1 &&
```

Comment:
After the previous check we know that node is defined, so we don't need the first `?.`. Also, we don't need it after `.typeParameters` in the next check.